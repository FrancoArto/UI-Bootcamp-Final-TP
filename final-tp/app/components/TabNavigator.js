import React from 'react';
import { createMaterialTopTabNavigator } from 'react-navigation'
import Icon from 'react-native-vector-icons/Feather'
import SecondIcon from 'react-native-vector-icons/EvilIcons'
import HomeScreen from '../screens/HomeScreen/HomeScreen'
import SearchScreen from '../screens/SearchScreen/SearchScreen';
import SettingsScreen from '../screens/SettingsScreen/SettingsScreen';


export default createMaterialTopTabNavigator({
  Forms:{ 
        screen: HomeScreen,
        navigationOptions: {
            tabBarIcon: ({ tintColor }) => (
                <Icon name="twitter" color={tintColor} size={24} />
            )
  }},
  Other:{ 
    screen: SearchScreen,
    navigationOptions: {
    tabBarIcon: ({tintColor}) => (
      <SecondIcon name="search" color={tintColor} size={32}/>
    )
        
  }},
  Settings: {
    screen: SettingsScreen,
    navigationOptions: {
      tabBarIcon: ({tintColor}) => (
        <Icon name="settings" color={tintColor} size={24}/>
      )
          
    }
  }   
},{ tabBarOptions: { showIcon: true, showLabel: false }})