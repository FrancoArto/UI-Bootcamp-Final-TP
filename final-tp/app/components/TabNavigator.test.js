import React from 'react';

import ShallowRenderer from 'react-test-renderer/shallow';
import TabNavigator from './TabNavigator';


it('renders without crashing', () => {
  const rendered = new ShallowRenderer()
  rendered.render(<TabNavigator />)
  expect(rendered).toMatchSnapshot();
});