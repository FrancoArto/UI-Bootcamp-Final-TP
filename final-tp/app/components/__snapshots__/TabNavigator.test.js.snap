// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_element": <NavigationContainer />,
  "_forcedUpdate": false,
  "_instance": NavigationContainer {
    "_actionEventSubscribers": Set {},
    "_getScreenProps": [Function],
    "_handleOpenURL": [Function],
    "_initialAction": Object {
      "type": "Navigation/INIT",
    },
    "_navigation": Object {
      "actions": Object {},
      "addListener": [Function],
      "dangerouslyGetParent": [Function],
      "dispatch": [Function],
      "getChildNavigation": [Function],
      "getScreenProps": [Function],
      "goBack": [Function],
      "isFocused": [Function],
      "navigate": [Function],
      "router": Object {
        "childRouters": Object {
          "Forms": null,
          "Other": null,
          "Settings": null,
        },
        "getActionCreators": [Function],
        "getActionForPathAndParams": [Function],
        "getComponentForRouteName": [Function],
        "getComponentForState": [Function],
        "getPathAndParamsForState": [Function],
        "getScreenOptions": [Function],
        "getStateForAction": [Function],
      },
      "setParams": [Function],
      "state": Object {
        "index": 0,
        "isTransitioning": false,
        "routes": Array [
          Object {
            "key": "Forms",
            "params": undefined,
            "routeName": "Forms",
          },
          Object {
            "key": "Other",
            "params": undefined,
            "routeName": "Other",
          },
          Object {
            "key": "Settings",
            "params": undefined,
            "routeName": "Settings",
          },
        ],
      },
    },
    "_persistNavigationState": [Function],
    "context": Object {},
    "dispatch": [Function],
    "props": Object {},
    "refs": Object {},
    "state": Object {
      "nav": Object {
        "index": 0,
        "isTransitioning": false,
        "routes": Array [
          Object {
            "key": "Forms",
            "params": undefined,
            "routeName": "Forms",
          },
          Object {
            "key": "Other",
            "params": undefined,
            "routeName": "Other",
          },
          Object {
            "key": "Settings",
            "params": undefined,
            "routeName": "Settings",
          },
        ],
      },
    },
    "subs": Object {
      "remove": [Function],
    },
    "updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
  },
  "_newState": null,
  "_rendered": <Navigator
    navigation={
      Object {
        "actions": Object {},
        "addListener": [Function],
        "dangerouslyGetParent": [Function],
        "dispatch": [Function],
        "getChildNavigation": [Function],
        "getScreenProps": [Function],
        "goBack": [Function],
        "isFocused": [Function],
        "navigate": [Function],
        "router": Object {
          "childRouters": Object {
            "Forms": null,
            "Other": null,
            "Settings": null,
          },
          "getActionCreators": [Function],
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        },
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "isTransitioning": false,
          "routes": Array [
            Object {
              "key": "Forms",
              "params": undefined,
              "routeName": "Forms",
            },
            Object {
              "key": "Other",
              "params": undefined,
              "routeName": "Other",
            },
            Object {
              "key": "Settings",
              "params": undefined,
              "routeName": "Settings",
            },
          ],
        },
      }
    }
  />,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
}
`;
